libs/sdk-ui-kit/src/DocumentHeader/DocumentHeader.tsx

```diff
@@ -20,16 +19,33 @@ function getEffectiveTitle(pageTitle: string, brandTitle: string): string {
 /**
  * @internal
  */
-const DocumentHeader: React.FC<IDocumentHeaderProps> = (props) => {
-    const { pageTitle = "", brandTitle = "", appleTouchIconUrl = "", faviconUrl = "" } = props;
-
-    return (
-        <Helmet>
-            <title>{getEffectiveTitle(pageTitle, brandTitle)}</title>
-            <link rel="apple-touch-icon" type="image/png" href={appleTouchIconUrl} />
-            <link rel="shortcut icon" type="image/x-icon" href={faviconUrl} />
-        </Helmet>
-    );
+const DocumentHeader: React.FC<IDocumentHeaderProps> = ({
+    pageTitle = "",
+    brandTitle = "",
+    appleTouchIconUrl = "",
+    faviconUrl = "",
+}) => {
+    useEffect(() => {
+        if (!document) return;
+
+        document.title = getEffectiveTitle(pageTitle, brandTitle);
+
+        const linkApple = (document.querySelector("link[rel='apple-touch-icon']") ||
+            document.createElement("link")) as HTMLLinkElement;
+        linkApple.rel = "apple-touch-icon";
+        linkApple.type = "image/png";
+        linkApple.href = appleTouchIconUrl;
+        document.head.appendChild(linkApple);
+
+        const linkFavicon = (document.querySelector("link[rel~='icon']") ||
+            document.createElement("link")) as HTMLLinkElement;
+        linkFavicon.rel = "shortcut icon";
+        linkFavicon.type = "image/x-icon";
+        linkFavicon.href = faviconUrl;
+        document.head.appendChild(linkFavicon);
```

Comment:
what if linkFavicon would be existing, provided by that querySelector? Please consider if we should still append in that case.